### PR TITLE
fix: add type check for packages/remix-netlify-edge

### PR DESCRIPTION
**Pull request by nickytonline**
Adds type checking for packages/remix-netlify-edge

Relates to #3104

- [ ] Docs
- [ ] Tests

Testing Strategy:

1. Clone this repository
2. Checkout the branch, e.g. `gh co 2`
3. Install all dependencies by running `yarn`
4. Run `node scripts/build.mjs --deno` from the root folder of the project in your shell.
5. All type checks should pass.

### CODE REVIEW

**Comment by nickytonline** [comment]
Currently I'm sorting our why it can't find `createRequestHandler` @ascorbic as it exist from the * export from deno in that file.

```
➜ node scripts/build.mjs --deno                           
yarn run v1.22.19
$ /Users/nicktaylor/dev/work/remix/node_modules/.bin/rollup -c

packages/create-remix/cli.ts → build/node_modules/create-remix/dist...
created build/node_modules/create-remix/dist in 315ms

packages/remix/index.ts → build/node_modules/remix/dist...
created build/node_modules/remix/dist in 16ms

packages/remix/index.ts → build/node_modules/remix/dist/esm...
created build/node_modules/remix/dist/esm in 13ms

packages/remix-architect/index.ts → build/node_modules/@remix-run/architect/dist...
created build/node_modules/@remix-run/architect/dist in 77ms

packages/remix-cloudflare/index.ts → build/node_modules/@remix-run/cloudflare/dist...
created build/node_modules/@remix-run/cloudflare/dist in 50ms

packages/remix-cloudflare-pages/index.ts → build/node_modules/@remix-run/cloudflare-pages/dist/esm...
created build/node_modules/@remix-run/cloudflare-pages/dist/esm in 20ms

packages/remix-cloudflare-pages/index.ts → build/node_modules/@remix-run/cloudflare-pages/dist...
created build/node_modules/@remix-run/cloudflare-pages/dist in 22ms

packages/remix-cloudflare-workers/index.ts → build/node_modules/@remix-run/cloudflare-workers/dist/esm...
created build/node_modules/@remix-run/cloudflare-workers/dist/esm in 22ms

packages/remix-cloudflare-workers/index.ts → build/node_modules/@remix-run/cloudflare-workers/dist...
created build/node_modules/@remix-run/cloudflare-workers/dist in 21ms

packages/remix-deno/.empty.js → stdout...

(!) Generated an empty chunk
.empty

packages/remix-dev/index.ts → build/node_modules/@remix-run/dev/dist...
created build/node_modules/@remix-run/dev/dist in 568ms

packages/remix-dev/cli.ts → build/node_modules/@remix-run/dev/dist...
created build/node_modules/@remix-run/dev/dist in 23ms

packages/remix-dev/server-build.ts → build/node_modules/@remix-run/dev, build/node_modules/@remix-run/dev/dist...
created build/node_modules/@remix-run/dev, build/node_modules/@remix-run/dev/dist in 31ms

packages/remix-express/index.ts → build/node_modules/@remix-run/express/dist...
created build/node_modules/@remix-run/express/dist in 21ms

packages/remix-netlify/index.ts → build/node_modules/@remix-run/netlify/dist...
created build/node_modules/@remix-run/netlify/dist in 21ms

packages/remix-netlify-edge/index.ts → build/node_modules/@remix-run/netlify-edge/dist...
created build/node_modules/@remix-run/netlify-edge/dist in 16ms

packages/remix-node/index.ts → build/node_modules/@remix-run/node/dist...
created build/node_modules/@remix-run/node/dist in 61ms

packages/remix-react/index.tsx → build/node_modules/@remix-run/react/dist...
created build/node_modules/@remix-run/react/dist in 418ms

packages/remix-react/index.tsx → build/node_modules/@remix-run/react/dist/esm...
created build/node_modules/@remix-run/react/dist/esm in 284ms

packages/remix-serve/index.ts, packages/remix-serve/env.ts → build/node_modules/@remix-run/serve/dist...
created build/node_modules/@remix-run/serve/dist in 11ms

packages/remix-serve/cli.ts → build/node_modules/@remix-run/serve/dist...
created build/node_modules/@remix-run/serve/dist in 11ms

packages/remix-server-runtime/index.ts → build/node_modules/@remix-run/server-runtime/dist...
created build/node_modules/@remix-run/server-runtime/dist in 263ms

packages/remix-server-runtime/index.ts → build/node_modules/@remix-run/server-runtime/dist/esm...
created build/node_modules/@remix-run/server-runtime/dist/esm in 241ms

packages/remix-vercel/index.ts → build/node_modules/@remix-run/vercel/dist...
created build/node_modules/@remix-run/vercel/dist in 16ms
Done in 3.02s.
Import map diagnostics:
  - Invalid top-level key "comment". Only "imports" and "scopes" can be present.
Download https://esm.sh/v99/@types/mime@3.0.1/Mime
Check file:///Users/nicktaylor/dev/work/remix/packages/remix-netlify-edge/index.ts
error: TS2614 [ERROR]: Module '"https://esm.sh/@remix-run/deno@0.0.0-nightly-a79f828-20221130"' has no exported member 'createRequestHandler'. Did you mean to use 'import createRequestHandler from "https://esm.sh/@remix-run/deno@0.0.0-nightly-a79f828-20221130"' instead?
import { createRequestHandler as createRemixRequestHandler } from "@remix-run/deno";
         ~~~~~~~~~~~~~~~~~~~~
    at file:///Users/nicktaylor/dev/work/remix/packages/remix-netlify-edge/server.ts:1:10

TS2614 [ERROR]: Module '"https://esm.sh/@remix-run/deno@0.0.0-nightly-a79f828-20221130"' has no exported member 'ServerBuild'. Did you mean to use 'import ServerBuild from "https://esm.sh/@remix-run/deno@0.0.0-nightly-a79f828-20221130"' instead?
import type { ServerBuild } from "@remix-run/deno";
              ~~~~~~~~~~~
    at file:///Users/nicktaylor/dev/work/remix/packages/remix-netlify-edge/server.ts:2:15

Found 2 errors.
Error: deno exited with code 1
    at ChildProcess.<anonymous> (file:///Users/nicktaylor/dev/work/remix/scripts/build.mjs:48:16)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1100:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:304:5)
```

File: scripts/build.mjs [hunk 1]
```diff
@@ -15,6 +15,7 @@ exec("yarn", ["rollup", "-c"])
         "check",
         "--import-map=.vscode/deno_resolve_npm_imports.json",
         ...glob.sync("packages/remix-deno/**/*.ts"),
+        ...glob.sync("packages/remix-netlify-edge/**/*.ts"),
```
Comment (ascorbic) [inline scripts/build.mjs:18]:
```suggestion
        ...glob.sync("packages/{remix-deno,remix-netlify-edge}/**/*.ts"),
```
These can be combined.

**Review by ascorbic — APPROVED** [review]
Looking again at the Deno handler, it's getting us in a bit of a mess. The signature doesn't allow you to pass in the context, and once it calls `getLoadContext` the context object isn't in scope. I think the simplest solution is to call the version of `createRequestHandler` from server-runtime directly. Rather than importing it from there, we could re-export it as `createRemixRequestHandler` from `@remix-run/deno`. That then allows us to pass in the context directly.

**Comment by nickytonline** [comment]
> Looking again at the Deno handler, it's getting us in a bit of a mess. The signature doesn't allow you to pass in the context, and once it calls `getLoadContext` the context object isn't in scope. I think the simplest solution is to call the version of `createRequestHandler` from server-runtime directly. Rather than importing it from there, we could re-export it as `createRemixRequestHandler` from `@remix-run/deno`. That then allows us to pass in the context directly.

Thanks @ascorbic, I was toying with using it from `server-runtime` directly yesterday for the reasons you mentioned, so sounds like I'm on the right track.

**Comment by nickytonline** [comment]
Still sorting out some type issues:

```
error: Module not found "file:///Users/nicktaylor/dev/work/remix/packages/remix-netlify-edge/node_modules/@remix-run/server-runtime/dist/routes".
    at file:///Users/nicktaylor/dev/work/remix/packages/remix-netlify-edge/node_modules/@remix-run/server-runtime/dist/build.d.ts:3:42
Error: deno exited with code 1
    at ChildProcess.<anonymous> (file:///Users/nicktaylor/dev/work/remix/scripts/build.mjs:47:16)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1100:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:304:5)
```

File: packages/remix-deno/server.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import * as path from "https://deno.land/std@0.128.0/path/mod.ts";
 import mime from "mime";
-import { createRequestHandler as createRemixRequestHandler } from "@remix-run/server-runtime";
+import { createRequestHandler as _createRemixRequestHandler } from "@remix-run/server-runtime";
```
Comment (ascorbic) [inline packages/remix-deno/server.ts:3]:
```suggestion
import { createRequestHandler as createRemixRequestHandler } from "@remix-run/server-runtime";
export { createRemixRequestHandler }
```

**Comment by ascorbic** [comment]
The error is because you need to update the versions in the package.json. Right now because the versions don't match it's installing them from npm instead of the version from the monorepo.

File: packages/remix-deno/server.ts (hunk 2 above)
Comment (nickytonline) [reply packages/remix-deno/server.ts]: For some reason my brain was stuck on the re-export syntax yesterday. Thanks!

**Pull request merged**